### PR TITLE
Fixing git ai issues

### PR DESCRIPTION
**Pull request by St4rman**
This PR is for enemies being damageable and some added states.

* The IDamageable is an interface that you can inherit from and that has the latent TakeDamage(int damage) function that both players and enemies can use. EStats is the new enemyStats script and PlayerStats is a debug script i made rn to test player damage to enemies
* Float UI and State UI are for debugs, and This branch also has player being chased 
* Press H to kill an enemy slowly but surely

### CODE REVIEW

File: quirklike/Assets/AI/FSM/States/Wanderer Specific/IdleState.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    private int IDLE_TIME = 1000;
+    
+    public IdleState(Wanderer w, NavMeshAgent a){
+        this.wanderer = w;
+        this.NavAgent = a;
+    }
+
+    public void Tick()
+    { 
+        IDLE_TIME --;
```
Comment (DelCherub) [inline quirklike/Assets/AI/FSM/States/Wanderer Specific/IdleState.cs:17]: Won't this be framerate dependent? 1 tick = 1 Update() so this should probably be a float instead and use deltaTime?

File: quirklike/Assets/AI/Standard Scripts/EnemyStats.cs [hunk 2]
```diff
@@ -6,22 +6,28 @@
 //the future but do i care? yes 
 //but it is a problem for future me to worry about
 
-public class EnemyStats : MonoBehaviour {
+public class EnemyStats : MonoBehaviour, IDamageable
```
Comment (DelCherub) [inline quirklike/Assets/AI/Standard Scripts/EnemyStats.cs]: What is the difference between EStats and EnemyStats? like what purpose do they each have

**Review by DelCherub — CHANGES_REQUESTED** [review]
Looks alright but just kinda want some clarity on a few things :)

File: quirklike/Assets/AI/FSM/States/Wanderer Specific/IdleState.cs (hunk 1 above)
Comment (St4rman) [reply quirklike/Assets/AI/FSM/States/Wanderer Specific/IdleState.cs:17]: This is mostly debug ngl, we can use deltaTime in the actual build :] the idle is just there to have the wake up animation tbh with you. But yeah can change it to detla time if really required

File: quirklike/Assets/AI/Standard Scripts/EnemyStats.cs (hunk 2 above)
Comment (St4rman) [reply quirklike/Assets/AI/Standard Scripts/EnemyStats.cs]: Enemystats is not suppsoed to be there. Estats is the new version. I tried changing Enemystats but decided to write a new script instead 


Estats is basically supposed to be the generalized stats script that you can stick onto any enemy and then interface with the TakeDamage function, and getposition if you need it. If you take a look at playerstats (also an example rn for debug) you can see how it looks for estats and checks if it has an Idamageable interface. Then all you gotta do is call take damage and itll do said damage

**Pull request merged**